test/libsolidity/SemanticTest.cpp

```diff
@@ -106,169 +108,223 @@ TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePref
+			}
+			AnsiColorized(_stream, _formatted, {BOLD, RED})
+				<< _linePrefix << endl
+				<< _linePrefix << "Attention: Updates on the test will apply the detected format displayed." << endl;
+			if (_compileViaYul && m_runWithoutYul)
+			{
+				_stream << _linePrefix << endl << _linePrefix;
+				AnsiColorized(_stream, _formatted, {RED_BACKGROUND}) << "Note that the test passed without Yul.";
+				_stream << endl;
+			}
+			else if (!_compileViaYul && m_runWithYul)
+				AnsiColorized(_stream, _formatted, {BOLD, YELLOW})
+					<< _linePrefix << endl
+					<< _linePrefix << "Note that the test also has to pass via Yul." << endl;
+			return TestResult::Failure;
 		}
 	}
-
-	if (!m_runWithYul && _compileViaYul)
+	catch (WhiskersError const&)
 	{
-		m_compileViaYulCanBeSet = success;
-		string message = success ?
-			"Test can pass via Yul, but marked with \"compileViaYul: false.\"" :
-			"Test compiles via Yul, but it gives different test results.";
-		AnsiColorized(_stream, _formatted, {BOLD, success ? YELLOW : MAGENTA}) <<
-			_linePrefix << endl <<
-			_linePrefix << message << endl;
-		return TestResult::Failure;
+		// this is an error in Whiskers template, so should be thrown anyway
+		throw;
 	}
-
-	if (!success && (m_runWithYul || !_compileViaYul))
+	catch (YulException const&)
 	{
-		AnsiColorized(_stream, _formatted, {BOLD, CYAN}) << _linePrefix << "Expected result:" << endl;
-		for (auto const& test: m_tests)
-		{
-			ErrorReporter errorReporter;
-			_stream << test.format(errorReporter, _linePrefix, false, _formatted) << endl;
-			_stream << errorReporter.format(_linePrefix, _formatted);
-		}
-		_stream << endl;
-		AnsiColorized(_stream, _formatted, {BOLD, CYAN}) << _linePrefix << "Obtained result:" << endl;
-		for (auto const& test: m_tests)
-		{
-			ErrorReporter errorReporter;
-			_stream << test.format(errorReporter, _linePrefix, true, _formatted) << endl;
-			_stream << errorReporter.format(_linePrefix, _formatted);
-		}
-		AnsiColorized(_stream, _formatted, {BOLD, RED})
-			<< _linePrefix << endl
-			<< _linePrefix << "Attention: Updates on the test will apply the detected format displayed." << endl;
-		if (_compileViaYul && m_runWithoutYul)
-		{
-			_stream << _linePrefix << endl << _linePrefix;
-			AnsiColorized(_stream, _formatted, {RED_BACKGROUND}) << "Note that the test passed without Yul.";
-			_stream << endl;
-		}
-		else if (!_compileViaYul && m_runWithYul)
-			AnsiColorized(_stream, _formatted, {BOLD, YELLOW})
-				<< _linePrefix << endl
-				<< _linePrefix << "Note that the test also has to pass via Yul." << endl;
-		return TestResult::Failure;
+		// this should be an error in yul compilation or translation
+		throw;
+	}
+	catch (boost::exception const&)
+	{
+		if (!_compileViaYul || m_runWithYul)
+			throw;
+	}
+	catch (std::exception const&)
+	{
+		if (!_compileViaYul || m_runWithYul)
+			throw;
+	}
+	catch (...)
+	{
+		if (!_compileViaYul || m_runWithYul)
+			throw;
 	}
```

Comment:
Why do we need this new block of catches?

Reply:
Oha! Looks like that I forgot to remove it. It is not needed. Not sure why I was putting it here in the first place. Great that you noticed!